framework/include/multiapps/MultiApp.h

```diff
@@ -138,6 +138,11 @@ class MultiApp : public MooseObject,
    */
   virtual void initialSetup() override;
 
+  /**
+   * Method which can be used for the reinitialization of the multiapp.
+   */
+  virtual void reinitialize() {}
```

Comment:
add a mooseError("not implemented")
otherwise people will call this on a transientApp and think they are good to go